### PR TITLE
Modified filter checks for media_type and per_page parameters

### PR DESCRIPTION
**Pull request by SID177**
The filter and sanitization on `media_type` and `per_page` parameters was not working. I changed it accordingly in order to work properly.

Issue: #1099

### CODE REVIEW

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 1]
```diff
@@ -1065,15 +1065,23 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
-			} else {
-				$media_type = $media_type_temp;
-			}
-			//Check array for currently allowed media types
-			$media_type = array_intersect( $media_type, $allowed_types );
-		}
+		if ( ! empty( $media_type_temp ) ) {
+			/**
+			 * Explode by , when string found
+			 */
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Use `//` comment style instead.

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 2]
```diff
@@ -1065,15 +1065,23 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+			$media_type = explode( ',', $media_type_temp );
+		} elseif ( ! empty( $media_type_array ) && is_array( $media_type_array ) ) {
+			/**
+			 * Set media_type to passed array
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Use // comment style instead for inline comments.

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 3]
```diff
@@ -1065,15 +1065,23 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+			 */
+			$media_type = $media_type_array;
+		}
+		/**
+		 * Check array for currently allowed media types
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Use `//` comment style instead for inline comments.

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 4]
```diff
@@ -1065,15 +1065,23 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+		 */
+		$media_type = array_intersect( $media_type, $allowed_types );
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: If both `if else if` condition not executed then `$media_type` will throw a notice. no?

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 5]
```diff
@@ -1116,12 +1124,18 @@ function rtmedia_api_process_rtmedia_gallery_request() {
 				$args['media_author'] = (int) $media_author;
 			}
 		}
-		$page     = filter_input( INPUT_POST, 'page', FILTER_SANITIZE_NUMBER_INT, FILTER_NULL_ON_FAILURE );
-		$per_page = filter_input( INPUT_POST, 'per_page', FILTER_SANITIZE_NUMBER_INT, FILTER_NULL_ON_FAILURE );
-		$order_by = filter_input( INPUT_POST, 'order_by', FILTER_SANITIZE_STRING, FILTER_NULL_ON_FAILURE );
+		/**
+		 * FILTER_NULL_ON_FAILURE is only for boolean values, using it will return FALSE only.
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Use `//` comment style instead for inline comments.

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 6]
```diff
@@ -1116,12 +1124,18 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+		 */
+		$page     = filter_input( INPUT_POST, 'page', FILTER_SANITIZE_NUMBER_INT );
+		$per_page = filter_input( INPUT_POST, 'per_page', FILTER_SANITIZE_NUMBER_INT );
+		$order_by = filter_input( INPUT_POST, 'order_by', FILTER_SANITIZE_STRING );
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Also, use `sanitize_text_field()`

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 7]
```diff
@@ -1116,12 +1124,18 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
 
 		$offset   = ! empty( $page ) ? (int) $page : 0;
-		$per_page = ( isset( $per_page ) && ! is_null( $per_page ) ) ? (int) $per_page : 10;
+		/**
+		 * Removed is_null check, because it won't be null.
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Use `//` comment style instead for inline comments.

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 8]
```diff
@@ -1116,12 +1124,18 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+		 */
+		$per_page = ! empty( $per_page ) ? (int) $per_page : 10;
 		$order_by = ! empty( $order_by ) ? $order_by : 'media_id desc';
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Wrap inline condition withing `()`. `( ! empty( $order_by ) ) ? $order_by : 'media_id desc';`

File: app/main/controllers/api/RTMediaJsonApi.php [hunk 9]
```diff
@@ -1116,12 +1124,18 @@ function rtmedia_api_process_rtmedia_gallery_request() {
...
+		$page     = filter_input( INPUT_POST, 'page', FILTER_SANITIZE_NUMBER_INT );
+		$per_page = filter_input( INPUT_POST, 'per_page', FILTER_SANITIZE_NUMBER_INT );
+		$order_by = filter_input( INPUT_POST, 'order_by', FILTER_SANITIZE_STRING );
 
 		$offset   = ! empty( $page ) ? (int) $page : 0;
-		$per_page = ( isset( $per_page ) && ! is_null( $per_page ) ) ? (int) $per_page : 10;
+		/**
+		 * Removed is_null check, because it won't be null.
+		 */
+		$per_page = ! empty( $per_page ) ? (int) $per_page : 10;
```
Comment (chandrapatel) [inline app/main/controllers/api/RTMediaJsonApi.php]: Wrap inline condition withing `()`. `( ! empty( $per_page ) ) ? (int) $per_page : 10;`

File: app/main/controllers/api/RTMediaJsonApi.php (hunk 4 above)
Comment (SID177) [reply app/main/controllers/api/RTMediaJsonApi.php]: Default `$media_type` is defined before, so if both is not executed, the default will be processed.

**Pull request merged**